src/ReduxStore/reducers/currentCity.js

```diff
@@ -0,0 +1,50 @@
+import {
+  CURRENT_CITY_LOADED,
+  CURRENT_CITY_LOADING,
+  CURRENT_CITY_FAIL,
+  SET_CURRENT_CITY_NAME
+} from '../actions/currentCity';
+
+const initialState = {
+  city: {},
+  isLoaded: false,
+  error: null,
+  cityName: ''
+};
+
+const currentCity = (state = initialState, action) => {
+  switch (action.type) {
+    case CURRENT_CITY_LOADING: {
+      return {
+        ...state,
+        isLoaded: false
+      };
+    }
+    case CURRENT_CITY_LOADED: {
+      return {
+        ...state,
+        isLoaded: true,
+        city: action.payload
+      };
+    }
+    case CURRENT_CITY_FAIL: {
```

Comment:
Honestly saying I don't like this name. It is not your city that failed, but the loading of it's data. So it is better to name this action `CURRENT_CITY_LOADING_FAILED`.... or similar.